src/Components/UpcomingTable/UpcomingTable.tsx

```diff
@@ -112,6 +112,27 @@ export const UpcomingTable: React.FunctionComponent<UpcomingTableProps> = ({
     return currentData.sort((a, b) => {
       const aValue = getSortableRowValues(a)[index];
       const bValue = getSortableRowValues(b)[index];
+
+      // Special handling for release column (index 2) - semantic version sorting
+      if (index === 2 && typeof aValue === 'string' && typeof bValue === 'string') {
+        const parseVersion = (version: string) => {
+          const parts = version.split('.').map((part) => parseInt(part, 10) || 0);
+          return { major: parts[0] || 0, minor: parts[1] || 0 };
+        };
+
+        const versionA = parseVersion(aValue);
+        const versionB = parseVersion(bValue);
+
+        // Compare major versions first
+        if (versionA.major > versionB.major) return direction === 'asc' ? 1 : -1;
+        if (versionA.major < versionB.major) return direction === 'asc' ? -1 : 1;
+        // If major versions are equal, compare minor versions
+        if (versionA.minor > versionB.minor) return direction === 'asc' ? 1 : -1;
+        if (versionA.minor < versionB.minor) return direction === 'asc' ? -1 : 1;
+        return 0;
```

Comment:
```suggestion
        const aNum = parseFloat(aValue);
        const bNum = parseFloat(bValue);
        const comparison = aNum - bNum;
        return direction === 'asc' ? comparison : -comparison;
```

We could use a [parseFloat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat) there and reduce the complexity of the code. I tested that and seems it works just fine.

Reply:
Works great Adam, thanks for the suggestion!